dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -520,6 +521,184 @@ private void handleEntry2tsv(List<String> args) throws ApiException, IOException
     }
 
     /**
+     * this function will check if the content of the file is CWL or not
+     * @param content
+     * @return
+     */
+    private Boolean checkCWL(File content){
+        /* CWL: check for 'class:Workflow OR CommandLineTool', 'inputs: ','outputs: ', and 'steps' */
+        Pattern cwlVersion = Pattern.compile("(.*)(cwlVersion)(.*)(:)(.*)");
+        Pattern classWfPattern = Pattern.compile("(.*)(class)(.*)(:)(\\sWorkflow)");
+        Pattern classToolPattern = Pattern.compile("(.*)(class)(.*)(:)(\\sCommandLineTool)");
+        String line;
+        Boolean versionFound = false;
+
+        try{
+            InputStreamReader is = new InputStreamReader(new FileInputStream(content.getPath()),StandardCharsets.UTF_8);
+            BufferedReader bufferedReader = new BufferedReader(is);
+            while((line = bufferedReader.readLine()) != null) {
+                Matcher matchWf = classWfPattern.matcher(line);
+                Matcher matchTool = classToolPattern.matcher(line);
+                Matcher matchVersion = cwlVersion.matcher(line);
+                if(matchVersion.find() && !versionFound){
+                    versionFound = true;
+                } else {
+                    if(getEntryType().toLowerCase().equals("workflow") && matchWf.find() && versionFound){
+                        return true;
+                    } else if(getEntryType().toLowerCase().equals("tool") && matchTool.find() && versionFound){
+                        return true;
+                    }
+                    errorMessage("Entry type does not match the class specified in CWL file.",CLIENT_ERROR);
+                }
+            }
+            bufferedReader.close();
+        } catch (IOException e){
+            //e.printStackTrace();
```

Comment:
Please don't throw away exceptions. Either handle the error or throw a RuntimeException.